obspy/imaging/source.py

```diff
@@ -0,0 +1,595 @@
+        :meth:`obspy.core.event.event.Event.plot`.
+    :type subplot_size: float
+    :param subplot_size: Width/height of one single subplot cell in inches.
+    :rtype: tuple
+    :returns: A 3-tuple with a :class:`~matplotlib.figure.Figure`, a list of
+        :class:`~matplotlib.axes.Axes` and a list of strings with corresponding
+        plotting options for each axes (see
+        :meth:`obspy.core.event.event.Event.plot`, parameter `kind`).
+    """
+    # make 2d layout of kind parameter
+    if isinstance(kind[0], (list, tuple)):
+        nrows = len(kind)
+        ncols = max([len(k) for k in kind])
+    else:
+        nrows, ncols = 1, len(kind)
+        kind = [kind]
+    figsize = (ncols * subplot_size, nrows * subplot_size)
+    if not fig:
+        fig = plt.figure(figsize=figsize, facecolor='white')
+    # get tuple of "kind" options and corresponding axes
+    kind_ = []
+    axes = []
+    for i, row in enumerate(kind):
+        ncols_ = len(row)
+        for j, kind__ in enumerate(row):
+            kind_.append(kind__)
+            kwargs = {"aspect": "equal", "adjustable": "datalim"}
+            if kind__ in ("p_quiver", "p_sphere", "s_quiver", "s_sphere"):
+                kwargs["projection"] = "3d"
+            axes.append(fig.add_subplot(
+                nrows, ncols_, i * ncols_ + j + 1, **kwargs))
+    return fig, axes, kind_
+
+
+def plot_radiation_pattern(
+        mt, kind=['p_sphere', 'beachball'], coordinate_system='RTP',
+        p_sphere_direction='inwards', fig=None, show=True):
+    """
+    Plot the P/S farfield radiation pattern on a unit sphere grid.
+
+    The calculations are based on [Aki1980]_ eq. 4.29.
+
+    :param mt: Focal mechanism NM x 6 (M11, M22, M33, M12, M13, M23 - the
+        six independent components of the moment tensor, where the coordinate
+        system is 1,2,3 = Up,South,East which equals r,theta,phi -
+        Harvard/Global CMT convention). The relation to [Aki1980]_
+        x,y,z equals North,East,Down convention is as follows: Mrr=Mzz,
+        Mtt=Mxx, Mpp=Myy, Mrt=Mxz, Mrp=-Myz, Mtp=-Mxy.
+
+    :param kind: One of the following three options:
+        * A list of strings or nested list of strings for a matplotlib plot
+          (for details see :meth:`obspy.core.event.event.Event.plot`)
+        * 'mayavi': uses the mayavi library (not yet available under Python 3
+          and problematic with anaconda)
+        * 'vtk': This vtk option writes three vtk files to the current working
+          directory. rpattern.vtk contains the p and s wave farfield vector
+          field beachlines.vtk contains the nodal lines of the radiation
+          pattern rpattern.pvsm is a state file that sets ParaView parameters
+          to plot rpattern.vtk and beachlines.vtk
+
+    :param coordinate_system: the only implemented option so far is 'RTP'.
+                              Should be extended to support NED, USE, DSE, NED
```

Comment:
I'm re-adding this comment so that @MMesch can address it:
> Isn't RTP the same as USE? Why is this name used here, but not anywhere else (e.g., in MoPaD)?